clock_test.go

```diff
@@ -42,3 +46,31 @@ func TestWithClock(t *testing.T) {
 		assert.Equal(t, date, logs.All()[0].Entry.Time, "Unexpected entry time.")
 	})
 }
+
+func TestSystemClockNewTicker(t *testing.T) {
+	var target = int32(3)
+	var n atomic.Int32
+	done := make(chan struct{})
+	defer func() {
+		<-done
+		assert.Equal(t, target, n.Load())
+	}() // wait for end
+
+	quit := make(chan struct{})
+	// Create a channel to increment every millisecond.
+	go func(ticker *time.Ticker) {
+		defer close(done)
```

Comment:
Do we need this to be a separate goroutine?
For this one, since we're blocking until three ticks, we can just do this all inline. It would simplify book keeping and it doesn't need to be an atomic because there would be no concurrency:

```
target := 3
var n int

ticker := _systemClock.NewTicker(..)
for range ticker.C {
  n++
  if n == target {
    ticker.Stop()
    return
  }
}
```

(We don't need the `assert.Equal` either because if the logic is faulty, the function will just loop forever.)